src/Parser/Hunter/Survival/Modules/Spells/AzeriteTraits/WildernessSurvival.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import TraitStatisticBox, { STATISTIC_ORDER } from 'Interface/Others/TraitStatisticBox';
+import SPELLS from 'common/SPELLS/index';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+
+/**
+ * Raptor Strike (or Mongoose Bite) deals an additional 27 damage and reduces the remaining cooldown of Wildfire Bomb by 1.0 sec.
+ *
+ * Example report: https://www.warcraftlogs.com/reports/Xr7Nxjd1KnMT9QBf/#fight=1&source=13&type=summary
+ */
+
+const TRIGGERING_SPELLS = [
+  SPELLS.MONGOOSE_BITE_TALENT.id,
+  SPELLS.MONGOOSE_BITE_TALENT_AOTE.id,
+  SPELLS.RAPTOR_STRIKE.id,
+  SPELLS.RAPTOR_STRIKE_AOTE.id,
+];
+
+const MS_REDUCTION = 1000;
+
+class WildernessSurvival extends Analyzer {
+
+  static dependencies = {
+    spellUsable: SpellUsable,
+  };
+
+  effectiveWSReductionMs = 0;
+  wastedWSReductionMs = 0;
+  hasWFI = false;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.WILDERNESS_SURVIVAL.id);
+    if (this.selectedCombatant.hasTalent(SPELLS.WILDFIRE_INFUSION_TALENT.id)) {
+      this.hasWFI = true;
+    }
+  }
+
+  get effectiveCDRInSeconds() {
+    return this.effectiveWSReductionMs / 1000;
+  }
+
+  get wastedCDRInSeconds() {
+    return this.wastedWSReductionMs / 1000;
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    if (!TRIGGERING_SPELLS.includes(spellId)) {
+      return;
+    }
+    if (this.hasWFI) {
+      if (this.spellUsable.isOnCooldown(SPELLS.VOLATILE_BOMB_WFI.id)) {
+        if (this.spellUsable.cooldownRemaining(SPELLS.VOLATILE_BOMB_WFI.id) < MS_REDUCTION) {
+          const effectiveReductionMs = this.spellUsable.reduceCooldown(SPELLS.VOLATILE_BOMB_WFI.id, MS_REDUCTION);
+          this.effectiveWSReductionMs += effectiveReductionMs;
+          this.wastedWSReductionMs += (MS_REDUCTION - effectiveReductionMs);
+        } else {
+          this.effectiveWSReductionMs += this.spellUsable.reduceCooldown(SPELLS.VOLATILE_BOMB_WFI.id, MS_REDUCTION);
+        }
```

Comment:
You can avoid repeating code by moving this inner if/else block into a function, since all that changes in each check is the spell being checked. 
    
    checkCooldown(spell){
        if (this.spellUsable.cooldownRemaining(SPELLS.spell.id) < MS_REDUCTION) {
          const effectiveReductionMs = this.spellUsable.reduceCooldown(SPELLS.spell.id, MS_REDUCTION);
          this.effectiveWSReductionMs += effectiveReductionMs;
          this.wastedWSReductionMs += (MS_REDUCTION - effectiveReductionMs);
        } else {
          this.effectiveWSReductionMs += this.spellUsable.reduceCooldown(SPELLS.spell.id, MS_REDUCTION);
        }
    }